app/assets/javascripts/discourse/components/discourse-topic.js.es6

```diff
@@ -64,6 +64,24 @@ export default Ember.Component.extend(AddArchetypeClass, Scrolling, {
     this.appEvents.on('post:highlight', postNumber => {
       Ember.run.scheduleOnce('afterRender', null, highlight, postNumber);
     });
+
+    this.appEvents.on('header:update-topic', topic => {
+
+      if (topic === null) {
+        this._lastShowTopic = false;
+        this.appEvents.trigger('header:hide-topic');
+        return;
+      }
+
+      const offset = window.pageYOffset || $('html').scrollTop();
+      this._lastShowTopic = this.showTopicInHeader(topic, offset);
```

Comment:
I also do not that think that `offset` should be a parameter and it would be better if it were moved inside `showTopicInHeader`.